agent/structs/structs.go

```diff
@@ -262,6 +262,9 @@ type RegisterRequest struct {
 	// node portion of this update will not apply.
 	SkipNodeUpdate bool
 
+	// Check whether the request created from client agent
+	FromAgent bool
```

Comment:
Identifying the source of the request this way does work, but unfortunately this is a public interface, so the field could be used misused.